plutok-flat/src/main/kotlin/io/newm/plutok/uplc/builtin/BuiltinFunction.kt

```diff
@@ -0,0 +1,93 @@
+package io.newm.plutok.uplc.builtin
+
+/**
+ * All the possible builtin functions in Untyped Plutus Core.
+ */
+enum class BuiltinFunction(val index: Int) {
```

Comment:
Is the index value part of the spec? Just wondering if that's the reason to have an explicit `index` instead of just using the built-in the ordinal.

Reply:
either way should be fine. When encoded to Flat the builtin just needs to be able to translate to some Int. In rust I used an enum and explicitly gave the u8 value it should map too. So in Rust the enum member as a value is directly castable to some some u8. Considering that it could save some space to have a method for converting back and forth instead of carrying this extra Int around in memory.

Reply:
In the rust code this is based on, functions are organized by functionality instead of their index number. It seemed best to just specify the indexes the way the rust code did. That way, when they add new cryptographic functions with high index numbers, we can add them to the crypto section.

I believe the index numbers are used in the uplc code instead of the actual names of the functions.